cas/client.c

```diff
@@ -1823,9 +1827,11 @@ M0_INTERNAL int m0_cas_del(struct m0_cas_req *req,
 	M0_PRE(m0_cas_id_invariant(index));
 	M0_PRE(M0_IN(flags, (0, COF_DEL_LOCK, COF_SYNC_WAIT)));
 
-	(void)dtx;
 	rc = cas_req_prep(req, index, keys, NULL, keys->ov_vec.v_nr, flags,
 			  &op);
+	if (rc != 0)
+		return M0_ERR(rc);
+	rc = m0_dtx0_txd_copy(dtx, &op->cg_txd);
 	if (rc != 0)
 		return M0_ERR(rc);
```

Comment:
Do we need to do error handling here, e.g. to release the resources allocated in cas_req_prep()?

Reply:
No, it is done by the caller function. like dix_cas_rops_send():
----------------------------------------
		if (rc != 0) {
			m0_clink_del(&cas_rop->crp_clink);
			m0_clink_fini(&cas_rop->crp_clink);
			m0_cas_req_fini(&cas_rop->crp_creq);
			dix_cas_rop_rc_update(cas_rop, rc);
			cas_rop_tlink_del_fini(cas_rop);
			dix_cas_rop_fini(cas_rop);
			m0_free(cas_rop);
		}
----------------------------------------